lang_java/parsing/CONTRIBUTING.md

```diff
@@ -0,0 +1,56 @@
+# External Java Parsers
+
+We currently have WIP for incorporating tree-sitter-java grammar into pfff. 
+
+### Writing OCaml
+
+To compile OCaml code
+
+```
+$ ocamlc -o <executable> <filename>.ml
+$ ./<executable>
+```
+
+### Running
+
+You can run the following commands to output a JSON of the AST parsed from a `.java` file by tree-sitter.
+
+```
+$ npm install tree-sitter
+$ npm install tree-sitter-java
+$ git clone https://github.com/tree-sitter/tree-sitter-java/
```

Comment:
what is the difference between npm install tree-sitter-java and the git clone and later npm install?
You need the very last version of tree-sitter-java? The one in npm is too old? Then why do you need to install it?

Reply:
I haven't diff'ed the code on npm vs in the repository, I just remember that I needed to manually run npm install on the repository